private/buf/bufsync/syncer.go

```diff
@@ -384,28 +392,24 @@ func (s *syncer) syncModuleInBranch(
 		if builtModule == nil {
 			return fmt.Errorf("syncable commit %s has no built module to sync", commitHash)
 		}
-		if builtModule.ModuleIdentity() == nil {
-			return fmt.Errorf("built module for commit %s has no module identity", commitHash)
-		}
-		modIdentity := builtModule.ModuleIdentity().IdentityString()
```

Comment:
This was redundant, all modules in the syncable commits slice had the same identities because they were being renamed to the target. We already have the target, so we can just don't rename anything, and use the target we already know.